slam-libraries/viam-cartographer/src/mapping/map_builder.cc

```diff
@@ -96,6 +96,20 @@ cartographer::sensor::TimedPointCloudData MapBuilder::GetDataFromFile(
 
 // TODO: There might still be a lot of room to improve accuracy & speed.
 // Might be worth investigating in the future.
+cartographer::transform::Rigid3d MapBuilder::GetLastGlobalPose(
+    int trajectory_id) {
+    if (GetLocalSlamResultPoses().size() > 0) {
+        auto latest_local_pose_ = GetLocalSlamResultPoses().back();
+        auto local_transform =
+            map_builder_->pose_graph()->GetLocalToGlobalTransform(
+                trajectory_id);
+
+        return local_transform * latest_local_pose_;
+    } else {
+        return cartographer::transform::Rigid3d();
+    }
```

Comment:
We shouldn't be returning the identity `Rigid3d` matrix here. This is the starting point of the robot within the map. If we for some reason wouldn't be able to find & return the most recent global pose, instead of setting it to (0,0,0) we should keep it at the latest value we found the robot to be. I suggest we raise an error here that indicates that we didn't find the most recent global pose.

Reply:
The return here is to ensure something is set in the function based off what the function has access to. The corresponding check in the slam service prevents the function from running if no local poses exist

Reply:
I can add a log for the error

Reply:
I don't think that's a good idea, because it can easily become a source of error to return a value that does not really exist. Meaning in this case, if there is no value to be found, then returning an identity value is incorrect, since there is technically no value to be returned. 

I would either raise an error if it can't find the latest pose or remove this function and leave the code where it originally was in the `CreateMap` function - which would solve this dilemma.

Reply:
kk I'll revert back